purchase_vendor_promotion/models/stock_warehouse_orderpoint.py

```diff
@@ -12,6 +12,24 @@ class StockWarehouseOrderpoint(models.Model):
     )
     promotion_date_start = fields.Date(compute="_compute_promotion", store=True)
     promotion_date_end = fields.Date(compute="_compute_promotion", store=True)
+    supplier_id = fields.Many2one(
+        compute="_compute_supplier_id", readonly=False, store=True
+    )
+
+    @api.depends("route_id", "route_id.force_vendor_with_best_promotion")
+    def _compute_supplier_id(self):
+        for rec in self:
+            if rec.route_id and rec.route_id.force_vendor_with_best_promotion:
+                suppliers = rec.product_id._prepare_sellers(False)
+                promotion_suppliers = suppliers.filtered(
+                    lambda x: x._is_promotion_active_or_upcoming()
+                )
+                if promotion_suppliers:
+                    rec.supplier_id = promotion_suppliers.sorted(key=lambda x: x.price)[
```

Comment:
Do you need to sort again by price ?  That was already done in `_prepare_sellers` no ?

Reply:
Hello @santostelmo, you're totally right. I wanted to prioritize the price at the beginning. But I think it's not necessary to repeat it.